services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -305,6 +305,18 @@ private boolean isDynamicUserProfile() {
         return session.getProvider(UserProfileProvider.class).getConfiguration() != null;
     }
 
+    /**
+     * Get sure that correct hostname and path is used for totp form.
+     * Relevant when running in proxy mode.
+     *
+     * @return UriBuilder with configured hostname and path set
+     */
+    private UriBuilder getTotpUriBuilder() {
+        return uriInfo.getBaseUriBuilder()
+                .path(uriInfo.getRequestUri().getPath())
```

Comment:
Try changing to `replacePath`. Otherwise, the resulting URL would include two occurrences of `/auth`.

Reply:
Thanks @pedroigor. I am going to try this.

Reply:
@pedroigor I applied your suggested change. Seems to work as expected :)